bfs_battle_for_supremacy/game_logic/managers/cards_manager.py

```diff
@@ -16,20 +13,14 @@ class CardsManager:
     @staticmethod
     def provide_card(player):
         CardsManager.loading = True
-
         card_data = AiHandler.send_card_request()
 
         card_type = card_data.get("type")
         if not card_type:
             CardsManager.loading = False
             return None
 
-        if card_type == "monster":
-            card = Monster(**card_data)
-        elif card_type == "building":
-            card = Building(**card_data)
-        else:
-            card = Card(**card_data)
+        card = Card(**card_data)
```

Comment:
I think you should delete Monster and Building classes if you chose this approach.

(there is no use of Monster or Building Class in the entire code now except for testing files and this is totally wrong PLEASE REMOVE TESTS IF NOT DONE CORRECTLY).